src/rust/en.mangamonks/src/lib.rs

```diff
@@ -0,0 +1,151 @@
+		"Popular" => format!("{}/popular-manga/{}", BASE_URL, page),
+		"Admin's Choices" => format!("{}/admins-choices/{}", BASE_URL, page),
+		"Most Viewed" => format!("{}/most-viewed/{}", BASE_URL, page),
+		_ => format!("{}/new-arrivals/{}", BASE_URL, page),
+	};
+	let html = Request::new(url, HttpMethod::Get).html()?;
+	parse_manga_listing(html)
+}
+
+#[get_manga_list]
+fn get_manga_list(filters: Vec<Filter>, page: i32) -> Result<MangaPageResult> {
+	let mut included_tags: Vec<String> = Vec::new();
+	let mut excluded_tags: Vec<String> = Vec::new();
+	let mut status: String = String::new();
+	let mut title: String = String::new();
+	let mut manga_type: String = String::new();
+	let status_options = ["", "ongoing", "completed"];
+	let type_options = ["", "japanese", "korean", "chinese"];
+	for filter in filters {
+		match filter.kind {
+			FilterType::Title => {
+				title = filter.value.as_string()?.read();
+			}
+			FilterType::Genre => match filter.value.as_int().unwrap_or(-1) {
+				0 => excluded_tags.push(filter.object.get("id").as_string()?.read()),
+				1 => included_tags.push(filter.object.get("id").as_string()?.read()),
+				_ => continue,
+			},
+
+			FilterType::Select => {
+				let index = filter.value.as_int().unwrap_or(-1) as usize;
+				match filter.name.as_str() {
+					"Status" => status = String::from(status_options[index]),
+					"Type" => manga_type = String::from(type_options[index]),
+					_ => continue,
+				}
+			}
+			_ => continue,
+		};
+	}
+
+	if !title.is_empty() {
+		let parameters = format!("dataType=json&phrase={}", title);
+		let url = format!("{}/search/live", BASE_URL);
+		let json = Request::new(url, HttpMethod::Post)
+			.header("User-Agent", "Aidoku")
+			.header(
+				"Content-Type",
+				"application/x-www-form-urlencoded; charset=UTF-8",
+			)
+			.header("Accept", "application/json")
+			.header("X-Requested-With", "XMLHttpRequest")
+			.body(parameters)
+			.json()?
+			.as_object()?;
+		let results = json.get("manga").as_array()?;
+		let mut mangas: Vec<Manga> = Vec::new();
+		for manga in results {
+			if let Ok(manga_obj) = manga.as_object() {
+				let title = match manga_obj.get("title").as_string() {
+					Ok(node) => node.read(),
+					Err(_) => continue,
+				};
+				let id = match manga_obj.get("url").as_string() {
+					Ok(node) => node
+						.read()
+						.split('/')
+						.nth_back(0)
+						.unwrap_or_default()
+						.to_string(),
+					Err(_) => continue,
+				};
+				let cover = match manga_obj.get("image").as_string() {
+					Ok(node) => node.read(),
+					Err(_) => continue,
+				};
+				mangas.push(Manga {
+					id,
+					cover,
+					title,
+					..Default::default()
+				});
+			}
+		}
+		return Ok(MangaPageResult {
+			manga: mangas,
+			has_more: false,
+		});
+	}
+
+	let url = get_search_url(
+		BASE_URL,
+		included_tags,
+		excluded_tags,
+		manga_type,
+		status,
+		page,
+	);
+	let html = Request::new(url, HttpMethod::Get).html()?;
+	parse_manga_listing(html)
```

Comment:
no need for an explicit `return`, you can just have an else case and use implicit returns
```suggestion
		Ok(MangaPageResult {
			manga: mangas,
			has_more: false,
		})
	} else {
		let url = get_search_url(
			BASE_URL,
			included_tags,
			excluded_tags,
			manga_type,
			status,
			page,
		);
		let html = Request::new(url, HttpMethod::Get).html()?;
		parse_manga_listing(html)
	}
```